mitosheet/dev/create_new_step.py

```diff
@@ -0,0 +1,379 @@
+        {params_code}
+        return "TODO"
+
+    def get_code(self) -> List[str]:
+        {params_code}
+
+        # TODO: actually generate the code here!
+
+        return []
+
+    
+    def get_edited_sheet_indexes(self) -> List[int]:
+        return [] # TODO: return this here!
+    """
+
+def write_python_code_file(path_to_file: Path, code: str) -> None:
+
+    if os.path.exists(path_to_file):
+        clear = input(f'{path_to_file} already exists, do you want to clear it?: [y/n]').lower().startswith('y')
+        if clear:
+            os.remove(path_to_file)
+        else:
+            print("Ok... exiting")
+            exit(1)
+    
+    with open(path_to_file, 'w+') as f:
+        f.write(code)
+
+def write_step_performer(original_step_name: str, params: Dict[str, str]) -> None:
+    step_name = get_step_name(original_step_name)
+    path_to_file = get_step_performers_folder() / (step_name + '.py') 
+    step_performer_code = get_step_performer_code(original_step_name, params)
+    write_python_code_file(path_to_file, step_performer_code)
+
+def write_code_chunk(original_step_name: str, params: Dict[str, str]) -> None:
+    step_name = get_step_name(original_step_name)
+    path_to_file = get_code_chunk_folder() / (step_name + '_code_chunk.py') 
+    step_performer_code = get_code_chunk_code(original_step_name, params)
+    write_python_code_file(path_to_file, step_performer_code)
+
+def write_to_step_performer_init(original_step_name: str) -> None:
+    path_to_init = get_step_performers_folder() / '__init__.py'
+    (step_performer_name, step_performer_import_statement) = get_step_performer_name_and_import_statement(original_step_name) 
+
+    with open(path_to_init, 'r') as f:
+        code = f.read()
+        code = code.replace(IMPORT_MARKER, f"{step_performer_import_statement}\n{IMPORT_MARKER}")
+        code = code.replace(EXPORT_MARKER, f"{step_performer_name},\n    {EXPORT_MARKER}")
+
+    with open(path_to_init, 'w') as f:
+        f.write(code)
+
+def get_params_parameter_code(params: Dict[str, str]) -> str:
+    params_parameter_code = ""
+    for param_name, param_type in params.items():
+        # If we are working with column_ids, we turn them into headers
+        param_name = param_name.replace('_id', '_header')
+        param_type = param_type.replace("ColumnID", "ColumnHeader")
+
+        # NOTE: we turn column ids to column headers in the calling function for the test file
+        params_parameter_code += f'{param_name}: {param_type},\n            '
+    
+    return params_parameter_code.strip()
+
+def get_id_to_header_code(params: Dict[str, str]) -> str:
+    id_to_header_code = ""
+    for param_name, param_type in params.items():
+        if param_type == 'ColumnID':
+            id_to_header_code += f"""{param_name} =self.mito_widget.steps_manager.curr_step.column_ids.get_column_id_by_header(
+            sheet_index,
+            {param_name.replace('_id', '_header')}
```

Comment:
Wouldn't this only work if there have been no renames?

Reply:
Make sure that there is a test for this because we haven't generated any steps so far that actually rely on columns!